examples/kubernetes/README.md

```diff
@@ -0,0 +1,154 @@
+
+`sudo kubectl describe node minikube`
+Somewhere in capactity it should show nvidia.com/gpu: x 
+with x being number of available GPUs on our machine
+
+
+If everything worked correctlly we should be able to create a pod using a yaml template. Download a Dockerfile and build an image from it first. After that download the template, fill in a correct image name and run it. To see a detailed instrucion on how to run an example see README in deepspeech directory. 
+
+Template will create a traning job for Mozilla's Deepspeech using a single GPU.
+
+To output availavle nodes or pods use kubectl get pod/node.
+To see logs from a given node or pod use kubectl logs pod/node <name>.
+To remove pod use kubectl delete pod <name>.
+```
+sudo kubectl logs pod <ds>
+sudo kubectl describe pod <ds>
+sudo kubectl delete pod <ds>
+```
+
+
+
+## Running single process learning
+Template used above:
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: ds
+spec:
+  containers:
+    - args:
+      - python3
+      - ./DeepSpeech.py
+      - --train_files=ldc93s1/ldc93s1.csv
+      - --dev_files=ldc93s1/ldc93s1.csv
+      - --test_files=ldc93s1/ldc93s1.csv
+      - --log_level=0
+      - --train_batch_size=32
+      - --benchmark_steps=30
+      - --notest
+      name: ds-container
+      image: <yourdockerhub>/deepspeech
+      env:
+        - name: CUDA_VISIBLE_DEVICES
+          value: "0"
+      resources:
+        limits:
+          nvidia.com/gpu: 1
+  restartPolicy: Never
+```
+
+It provides a name for our pod, docker image source command with arguments to be run inside of the container, env variables and an amount of requested GPUs. 
+Even though we passed two GPUs to CUDA_VISIBLE_DEVICES it will not be able to "cheat" kubernetes and still will only have access to one GPU that was requested.
+
+
+## Running distributed learning
+There are two ways of running distributed training using kubernetes: deploying one container which has access to multiple GPUs and has multiple processes running inside of it or deploying multiple containers - one for each gpu with only one process running.
+
+### Summary
+Kubernetes can be very usefull in this age of containerization and cloud. It allows for an easy orchestration and automatization. However, it requires a lot of configuration besides that the whole ecosystem is quite new and in constant development which means things are changing fast! Its main focus is on cloud providers so local solutions are mainly for development and testing. Becuase of the fact how Docker handle cacheing development can be really slow. For example DeepSpeech's original Dockerfile contains building everything from scratch so after changing anything in code that can't be applied after this process rebuilding image takes hours. On top of that a popular option is to build and test images on different computer and that means uploading and downloading at least a few GB in case of rebuilding bigger layers.  
```

Comment:
1) Some typos
2) A little bit complicated

Kubernetes can be very useful in this age of containerization and cloud. However, it requires a significant amount of time and effort to configure on each machine. This whole ecosystem is quite new, under constant development which results in frequent changes and backward compatibility issues.

It's meant to be used by cloud providers, so local solutions are mainly restricted to development and testing. 
Additionally, considering how Docker handles layer caching, development can be really slow. For example DeepSpeech's original Dockerfile rebuilds everything from scratch when the code changes and that whole process can takes hours. 
An alternative way is to build and test images on different computers, but it leads to uploading and downloading at least a few GB in case of rebuilding **bigger / heavier** layers.